[sig-network] TestServiceRegistryUpdateDryRun flakes with "already allocated" error

**Which test(s) are flaking**:

TestServiceRegistryUpdateDryRun

**Reason for failure**:

```
--- FAIL: TestServiceRegistryUpdateDryRun (0.15s)
    rest_test.go:1305: Expected no error: Service "foo3" is invalid: spec.clusterIP: Invalid value: "1.2.3.4": provided IP is already allocated
```

This is flaking rarely enough that it is not caught by our CI jobs, which currently tolerate up to 2 unit test failures per run (!).

With that toleration removed in https://github.com/kubernetes/kubernetes/pull/93605, this flake has been seen (https://github.com/kubernetes/kubernetes/pull/93605#issuecomment-668142138).

Reproducible with the following steps:

1. Fix the bug in the test that casts the return value even when the create fails:

```diff
--- a/pkg/registry/core/service/storage/rest_test.go
+++ b/pkg/registry/core/service/storage/rest_test.go
@@ -1301,10 +1301,10 @@ func TestServiceRegistryUpdateDryRun(t *testing.T) {
                        }},
                },
        }, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-       svc = obj.(*api.Service)
        if err != nil {
                t.Fatalf("Expected no error: %v", err)
        }
+       svc = obj.(*api.Service)
        _, _, err = storage.Update(ctx, svc.Name, rest.DefaultUpdatedObjectInfo(&api.Service{
                ObjectMeta: metav1.ObjectMeta{
                        Name:            svc.Name,
```

2. Omit logging on slow etcd writes because of etcd-io/etcd#12197

```
--- a/vendor/go.etcd.io/etcd/etcdserver/apply_v2.go
+++ b/vendor/go.etcd.io/etcd/etcdserver/apply_v2.go
@@ -114,7 +114,7 @@ func (a *applierV2store) Sync(r *RequestV2) Response {
 // applyV2Request interprets r as a call to v2store.X
 // and returns a Response interpreted from v2store.Event
 func (s *EtcdServer) applyV2Request(r *RequestV2) Response {
-	defer warnOfExpensiveRequest(s.getLogger(), time.Now(), r, nil, nil)
+	// defer warnOfExpensiveRequest(s.getLogger(), time.Now(), r, nil, nil)
 
 	switch r.Method {
 	case "POST":
```

3. Build a test binary for the affected package:

```
go test -race -c ./pkg/registry/core/service/storage
```

4. Stress the affected test using the [stress](https://godoc.org/golang.org/x/tools/cmd/stress) tool:

```
stress ./storage.test -test.run TestServiceRegistryUpdateDryRun
```

5. Once failures are seen, look at the logs to see the details about the error:

```sh
ls -lat $TMPDIR/go-stress*
cat $TMPDIR/go-stress...<filename>
```

/sig network

--- a/pkg/registry/core/service/storage/rest_test.go
+++ b/pkg/registry/core/service/storage/rest_test.go
@@ -1182,10 +1182,10 @@ func TestServiceRegistryUpdateDryRun(t *testing.T) {
 			}},
 		},
 	}, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	svc := obj.(*api.Service)
 	if err != nil {
 		t.Fatalf("Expected no error: %v", err)
 	}
+	svc := obj.(*api.Service)
 
 	// Test dry run update request external name to node port
 	updatedSvc, created, err := storage.Update(ctx, svc.Name, rest.DefaultUpdatedObjectInfo(&api.Service{
@@ -1251,6 +1251,7 @@ func TestServiceRegistryUpdateDryRun(t *testing.T) {
 			Selector:        map[string]string{"bar": "baz"},
 			SessionAffinity: api.ServiceAffinityNone,
 			Type:            api.ServiceTypeNodePort,
+			ClusterIP:       "1.2.3.5",
 			Ports: []api.ServicePort{{
 				NodePort:   30020,
 				Port:       6502,
@@ -1259,10 +1260,11 @@ func TestServiceRegistryUpdateDryRun(t *testing.T) {
 			}},
 		},
 	}, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	svc = obj.(*api.Service)
 	if err != nil {
 		t.Fatalf("Expected no error: %v", err)
 	}
+	svc = obj.(*api.Service)
+
 	_, _, err = storage.Update(ctx, svc.Name, rest.DefaultUpdatedObjectInfo(&api.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:            svc.Name,
@@ -1301,10 +1303,10 @@ func TestServiceRegistryUpdateDryRun(t *testing.T) {
 			}},
 		},
 	}, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	svc = obj.(*api.Service)
 	if err != nil {
 		t.Fatalf("Expected no error: %v", err)
 	}
+	svc = obj.(*api.Service)
 	_, _, err = storage.Update(ctx, svc.Name, rest.DefaultUpdatedObjectInfo(&api.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:            svc.Name,

